llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -56,16 +57,12 @@ static bool parseRootSignatureElement(LLVMContext *Ctx,
 
   RootSignatureElementKind ElementKind =
       StringSwitch<RootSignatureElementKind>(ElementText->getString())
-          .Case("RootFlags", RootSignatureElementKind::RootFlags)
-          .Default(RootSignatureElementKind::None);
+          .Case("RootFlags", RootSignatureElementKind::RootFlags);
```

Comment:
What happens if the element kind is something unrecognized?  Something like:

```LLVM
!dx.rootsignatures = !{!2} ; list of function/root signature pairs
!2 = !{ ptr @main, !3 } ; function, root signature
!3 = !{ !4 } ; list of root signature elements
!4 = !{ !"I'mALittleTeapot", i32 1 } ; 
```

I don't see a test covering this case.